chaincfg/params.go

```diff
@@ -60,6 +61,37 @@ type DNSSeed struct {
 	HasFiltering bool
 }
 
+// ConsensusDeployment defines details related to a specific consensus rule
+// change that is voted in.  This is part of BIP0009.
+type ConsensusDeployment struct {
+	// BitNumber defines the specific bit number within the block version
+	// this particular soft-fork deployment refers to.
+	BitNumber uint8
+
+	// StartTime is the median block time after which voting on the
+	// deployment starts.
+	StartTime uint64
+
+	// ExpireTime is the median block time after which the attempted
+	// deployment expires.
+	ExpireTime uint64
+}
+
+// Constants that define the deployment offset in the deployments field of the
+// parameters for each deployment.  This is useful to be able to get the details
+// of a specific deployment by name.
+const (
+	// DeploymentTestDummy defines the rule change deployment ID for testing
+	// purposes.
+	DeploymentTestDummy = iota
+
+	// NOTE: MaxDeployments must always come last since it is used to
+	// determine how many defined deployments there currently are.
+
+	// MaxDeployments is how many defined deployments there currently are.
+	MaxDeployments
```

Comment:
Fair point.  It actually is used as a max at some places in the code as in `if deploymentID > ....`, however, that usage doesn't need to influence the name here and it wouldn't be unclear to read `if deploymentID > cfg.DefinedDeployments { ....`.

I'll change it to `DefinedDeployments` even though it's longer!